ihmc-perception/src/main/java/us/ihmc/perception/detections/yolo/YOLOv8Tools.java

```diff
@@ -77,30 +81,48 @@ public static double calculateStandardDeviationAndCentroid(List<? extends Point3
                                                               boolean shuffle,
                                                               Point3DBasics centroidToPack)
    {
+      if (pointCloud == null || pointCloud.isEmpty())
+         return 0.0;
+
       if (shuffle)
          Collections.shuffle(pointCloud);
 
-      Vector3D sumVector = new Vector3D(0.0, 0.0, 0.0);
-      Vector3D squaredSumVector = new Vector3D(0.0, 0.0, 0.0);
-      int numberOfSamples = Math.min(pointCloud.size(), maxNumberOfSamples);
+      int n = Math.min(pointCloud.size(), maxNumberOfSamples);
+
+      // Welford variables for each axis
+      double meanX = 0, meanY = 0, meanZ = 0;
+      double m2X = 0, m2Y = 0, m2Z = 0;
 
-      pointCloud.parallelStream().limit(numberOfSamples).forEach(point ->
+      // Welford's Algorithm
```

Comment:
Switched to using welfords algorithm for std dev calculation. the previous implementation was bugged and returned NaN all the time.